src/ParallelAlgorithms/Events/ObserveTime.hpp

```diff
@@ -0,0 +1,123 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/DataBox/TagName.hpp"
+#include "Domain/Tags.hpp"
+#include "IO/Observer/Helpers.hpp"
+#include "IO/Observer/ObservationId.hpp"
+#include "IO/Observer/ObserverComponent.hpp"  // IWYU pragma: keep
+#include "IO/Observer/ReductionActions.hpp"   // IWYU pragma: keep
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Parallel/Reduction.hpp"
+#include "ParallelAlgorithms/EventsAndTriggers/Event.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Tags.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Functional.hpp"
+#include "Utilities/Numeric.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace dg {
+namespace Events {
+template <typename ObservationValueTag, typename EventRegistrars>
+class ObserveTime;
+
+namespace Registrars {
+template <typename ObservationValueTag>
+using ObserveTime =
+    ::Registration::Registrar<Events::ObserveTime, ObservationValueTag>;
+}  // namespace Registrars
+
+template <typename ObservationValueTag,
+          typename EventRegistrars =
+              tmpl::list<Registrars::ObserveTime<ObservationValueTag>>>
+class ObserveTime;  // IWYU pragma: keep
+
+/*!
+ * \ingroup DiscontinuousGalerkinGroup
+ * \brief %Prints the time of observation in the simulation.
+ *
+ * Prints reduction quantities:
+ * - `ObservationValueTag`
+ *
+ * \warning Currently, only one reduction observation event can be
+ * triggered at a given observation value.  Causing multiple events to run at
+ * once will produce unpredictable results.
+ */
```

Comment:
Does this warning means I can't use this reduction event at the same time as say reducing analytic solution errors or constraints?